nucleus/__init__.py

```diff
@@ -462,13 +462,15 @@ def exception_handler(request, exception):
     def annotate_dataset(
         self,
         dataset_id: str,
-        annotations: List[DatasetItem],
+        annotations: List[Union[BoxAnnotation, PolygonAnnotation]],
+        update: bool,
```

Comment:
see comment on sister PR - I think this should be optional with default value False